src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Translate/Main.cs

```diff
@@ -0,0 +1,193 @@
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _iconPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            UpdateIconPath(Theme.Light);
+
+            _translator = new AggregateTranslator();
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// sd
+        /// </summary>
+        /// <param name="langs">Array of languages</param>
+        /// <returns>A converted list of 'language' to 'result'</returns>
+        public List<Result> ConvertArrayOfLanguagesToResult(Language[] langs)
+        {
+            var results = new List<Result>();
+
+            foreach (var lang in langs)
+            {
+                results.Add(new Result
+                {
+                    Title = lang.Code,
+                    SubTitle = $"{lang.DisplayName} - {lang.Name}",
+                    IcoPath = _iconPath,
+                    Action = (_) =>
+                    {
+                        Console.WriteLine("Action");
+                        return true;
+                    },
+                });
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            var arguments = query?.Search.Split(' ', 2);
+
+            var languageSearch = arguments?[0] ?? string.Empty;
+
+            var langs = LanguageHelper.FindLanguages(languageSearch);
+            var suggestedLangs = ConvertArrayOfLanguagesToResult(langs);
+
+            var targetText = (arguments?.Length > 1 && !string.IsNullOrWhiteSpace(arguments[1])) ? arguments[1] : null;
+            if (langs.Length > 0 && targetText is not null)
+            {
+                var translation = _translator.TranslateAsync(targetText, langs.First().Code);
```

Comment:
I believe all these services require a service key. How is this DLL making sure we respect their services revenue models.

Reply:
Yeah. Honestly, I have no idea.
I'm afraid the EULA might prevent this plugin from being released, but that's okay. I will use it anyway on all my devices.

Reply:
We “support” plugins and maybe that is a better way to distribute. A community member had a wsl distro and there is an everything search plugin too

Reply:
@crutkas
Can we list all this third party plugins somewhere that people can find them easily?